c2corg_api/security/roles.py

```diff
@@ -65,15 +65,21 @@ def create_claims(user, exp):
 
 
 def try_login(user, password, request):
-    if user.validate_password(password, DBSession):
-        policy = request.registry.queryUtility(IAuthenticationPolicy)
-        now = datetime.datetime.utcnow()
-        exp = now + datetime.timedelta(days=CONST_EXPIRE_AFTER_DAYS)
-        claims = create_claims(user, exp)
-        token = policy.encode_jwt(request, claims=claims).decode('utf-8')
-        return add_or_retrieve_token(token, exp, user.id)
+    if user.email_validated and user.validate_password(password):
+        return log_validated_user_i_know_what_i_do(user, request)
+    else:
+        return None
 
-    return None
+
+def log_validated_user_i_know_what_i_do(user, request):
```

Comment:
Yes, this is the safe `try_login` function, which take a (user, password) pair.
The `log_validated_user_i_dont_know_what_i_do` is a dangerous security risk if you did not validate the user, hence its name. Updating the documentation.